packages/components/src/FilterBar/FilterBar.component.js

```diff
@@ -120,6 +130,17 @@ export class FilterBarComponent extends React.Component {
 		if (this.props.onBlur) {
 			this.props.onBlur(event);
 		}
+		if (!this.state.value) {
+			this.props.onToggle(event);
+		}
+	}
+
+	onClear(event) {
+		if (this.props.onClear) {
```

Comment:
To avoid to play with manual focus, you can use `onMouseDown` and add a `e.preventDefault();`

```suggestion
		event.preventDefault();

		if (this.props.onClear) {
```